dask_cuda/device_host_file.py

```diff
@@ -10,14 +10,32 @@
 def _is_device_object(obj):
     """
     Check if obj is a device object, by checking if it has a
-    __cuda_array_interface__ attributed
+    __cuda_array_interface__ attribute or is instance of cudf.DataFrame
+    or cudf.Series.
     """
-    return hasattr(obj, "__cuda_array_interface__")
+    try:
+        import cudf
+        _device_instances = [cudf.DataFrame, cudf.Series]
+    except ImportError:
+        _device_instances = []
+    return (hasattr(obj, "__cuda_array_interface__") or
+            any([isinstance(obj, inst) for inst in _device_instances]))
```

Comment:
Importing `cudf` is non-trivially expensive (a few seconds).  Also, it would be nice not to have to special case libraries like this.  I wonder if there is another signal we can use in cases like these.  cc @kkraus14

Reply:
I do think though, that testing that cudf objects register as device objects would be very helpful. Same with lists of cudf objects (and other objects like cupy arrays and numba device arrays).

Reply:
Ping

Reply:
Thanks for the reminder. I agree that we would be better off not testing specifically for cudf objects, but currently there's no clean way of doing it. I've commented on this before as well https://github.com/rapidsai/dask-cuda/pull/65#issuecomment-497851259. For now, I'm checking those with `__module__` to avoid importing cudf.

Reply:
Once we extend `__cuda_array_interface__` with the mask attribute we'll be able to handle `cudf.Series` nicely, but `cudf.DataFrame` will still need to be special cased.

Given we'd need to special case any non-array-like GPU objects for this, i.e. an XGBoost DMatrix backed by device memory, it would be good if we could design a generalized approach.

Reply:
A generalized approach would be great, but AFAIK, there's no error-free way to check whether an object is backed by device memory or not. Do you know of any such way @kkraus14?

Reply:
He might be suggesting something like a type registry or dispatch function, maybe similar to how we handle the `sizeof` function today.

Reply:
This is a good idea. I've added that in the latest commit.